test/lint/lint-submodule.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+This script checks for git modules
+"""
+
+import subprocess
+
+def main():
+    submodule_check_args = ['git', 'submodule', 'status', '--recursive',]
+    try:
+        submodules_list = subprocess.check_output(submodule_check_args, stderr = subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        exit(1)
+
+    submodules_list = submodules_list.decode('utf-8').splitlines()
+
+    if submodules_list:
+        print("These submodules were found, delete them:")
+        for submodule in submodules_list:
+            print(submodule)
+        exit(1)
+    exit(0)
```

Comment:
I do have the same nit as in the other PR https://github.com/bitcoin/bitcoin/pull/24802#discussion_r846019598, if you use `exit`, use `sys.exit`.

Reply:
Thanks again @laanwj for the review! 
I've updated the PR.